Develop sign up functionality

BrandonVG opened the pull request:
# Description

## What we did
### Project change
This branch has a lot of changes but they are actually not real changes, we had a problem with the first generated project so we generated a new project with CLI and then just added the old changes we did before.

So please, ignore the changes in those files and focus on the functionalities changes and files that we have in /src
### Functionalities changes
We implemented new functional changes, using the Firebase connection we did in the past to make functional the sign-up screen, working to create users with email and password or using a Google account, we also added some validations to the form to only accept valid emails and passwords, also avoiding empty texts, for now, this signup just create the user in FireBase because we don't have another screen to redirect after this.

Finally, other functionalities like the password reveal on the icon eye pressed or the buttons being disabled until the user accepts the checkbox was added 

## Why we did 
### Project change
We needed to create a new base project because the first one gave us some problems with google sign in because of some SHA1 problems so by recreating the project we solved these problems, maybe this solution was not the best solution, but it worked well
### Function changes
We needed these changes to provide the user with the sign-in function and have the functionalities required in the requirements

## Screenshots

![image](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/54459465/c1bfc608-f50b-4476-8477-44914a930f32)


Co-authored-by: Tole15 <[rodrigermoon@gmail.com](mailto:rodrigermoon@gmail.com)>
Co-authored-by: Axw3ll <[axelomar.a.v@gmail.com](mailto:axelomar.a.v@gmail.com)>

Tole15 committed:
Separate Google Sign-In Logic into a Separate Hook File and Add Co-Designed Modal for Sign-In Stage

We have made some significant changes to enhance the sign-in functionality, specifically regarding the integration of Google sign-in and the introduction of a co-designed modal for the sign-in stage. These changes were made collaboratively by our team.

In this commit, we have taken the initiative to separate the logic related to Google sign-in into a dedicated hook file. By doing so, we aim to improve code organization and maintainability. This separation allows us to encapsulate the Google sign-in functionality within its own module, making it easier to manage and modify in the future. This step ensures a cleaner and more modular codebase.

Furthermore, we have added a co-designed modal for the sign-in stage. This modal provides a visually appealing and user-friendly interface for users to enter their credentials and authenticate with our system. By utilizing a co-designed approach, we have carefully crafted the modal's appearance to align with our application's overall design language, providing a consistent and cohesive user experience.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Fix bug in native login and improve password visibility icon

We've addressed a bug that affected the native login process, ensuring a smoother user experience. Additionally, we have corrected the aesthetic appearance of the password visibility icon, enhancing its functionality for password viewing. This commit reflects our efforts to improve the login functionality and user interface of the application.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

BrandonVG committed:
Fixed problems caused by rebase

While doing the rebase didnt accepted some changes so the <<<<HEAD tag was deleted

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

BrandonVG committed:
Used eslint

What we did?
We used eslint to make better our written code

Why we did?
We want to our code follow good practices and be easily to read following the eslint recommendations

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

BrandonVG committed:
Design functionality in the border color of field forms

Added the functionality to turn the input field border blue when it have text and change
it into gray when it dont have

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

BrandonVG committed:
Improved error shown in sign up screen

We change the way we show the errors on the screen putting it above the respective input
Also changed the handleSignup.js function to return the error to handle it and now the function
update with the first name

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Eduardoespl reviewed (approved):
💯

ecruz17 commented on .watchmanconfig:
```diff
@@ -1 +1 @@
-{}
\ No newline at end of file
+{}
```
nice

ecruz17 commented on src/hooks/SignInWithGoogle.js:
```diff
@@ -0,0 +1,27 @@
+import auth from '@react-native-firebase/auth';
+import {GoogleSignin} from '@react-native-google-signin/google-signin';
+
+GoogleSignin.configure({
+  scopes: ['email', 'profile'],
+  webClientId:
+    '569311043643-h9c0ctlutkqsnr2ticidfl3d08nuafqj.apps.googleusercontent.com',
```
you could place this inside an env variable

ecruz17 reviewed (approved):
Nice work! 🔥

BrandonVG committed:
Delete google-services.json

Deleted the file from the repository because we shouldn't have this file up here, but we uploaded it before adding it into the .gitignore


Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>
Co-authored-by: Tole15 <rodrigermoon@gmail.com>

BrandonVG committed:
Added .env to hide web client id key

Installed react-native-config package to use .env file and hide confidential info.

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
General Code and Component Improvements

Refactored code and optimized components for enhanced performance and maintainability. Made various improvements to enhance code readability, efficiency, and scalability. Resolved bugs and addressed code quality issues.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Updated SignUpButton and SignUpForm components

- Updated SignUpButton component to include a modal for displaying a small box on the screen.
- Modified SignUpForm component to use the UseFormState hook for managing form state.
- Made necessary changes to ensure the modal functionality works for both SignUp and SignUpWithGoogle.
- The modal design will be a small box displayed on the screen.

Note: The UseFormState.js file was added to the hooks folder to handle form state management.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Refactor Loading Modal into separated component, fix visualization bug, and improve design

Commit description:
- Refactored Loading Modal component to include separate modal component in a separate file for better organization.
- Worked on the design of the LoadingModal component to improve its visual appearance.
- Fixed a bug related to the visualization issue in the LoadingModal component.

Please note that the commit message and description provided above are for illustrative purposes and may need to be adjusted based on your project's conventions and requirements.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>
